model/registry/service.yaml

```
@@ -20,6 +20,12 @@ groups:
         type: string
         brief: >
           The version string of the service API or implementation. The format is not defined by these conventions.
+        note: |
+          When languages/platforms offer standard ways to version applications
+          and/or libraries SDKs MAY offer mechanisms to auto-populate `service.version`
+          from the standard sources. If SDKs offer an auto-population mechanism it MUST be
+          an optional feature users opt-into and if `service.version` is manually
+          populated it MUST take priority over auto-population.
```

> If SDKs offer an auto-population mechanism it MUST be
an optional feature users opt-into

any particular reason for this limitation? 

If it's back-compat only, then I think it's up to detector/SDK to decide. E.g. if detector/SDK is still in beta, they don't need to be backward compatible (or if detector is considering v2).

I'd remove or relax this limitation. 

So maybe something like:

```suggestion
        note: |
          When possible, service detector component MAY use language/platform-specific capabilities to determine application version and populate `service.version` automatically.
          
          If `service.version` is provided explicitly it MUST take priority over the version determined automatically.
```

?

---

I like the proposed version, but I agree with @lmolkova about the MUST. Maybe we can just say: (since it's optional anyway)

> SDKs may offer an opt-in mechanism to auto-populate `service.version` 

or something along those lines

---

This was all about backwards compatibility, ya. I updated to use the version provided by @lmolkova.